File: VB/Controllers/HomeController.vb
Imports Microsoft.VisualBasic
Imports DevExpress.Web.Mvc
Imports SchdulerBestPracticesMvc.Code
Imports SchdulerBestPracticesMvc.Models
Imports System.Web.Mvc

Namespace SchdulerBestPracticesMvc
	Public Class HomeController
		Inherits Controller
		Public Function Index() As ActionResult
			Return View(SchedulerDataHelper.DataObject)
		End Function

		Public Function SchedulerPartial() As ActionResult
			Return PartialView("SchedulerPartial", SchedulerDataHelper.DataObject)
		End Function

		Public Function EditAppointment() As ActionResult
			UpdateAppointment()
			Return PartialView("SchedulerPartial", SchedulerDataHelper.DataObject)
		End Function

		Private Shared Sub UpdateAppointment()
			Dim insertedAppt As CarScheduling = SchedulerExtension.GetAppointmentToInsert(Of CarScheduling)(SchedulerHelper.Settings, SchedulerDataHelper.GetAppointments(), SchedulerDataHelper.GetResources())
			SchedulerDataHelper.InsertAppointment(insertedAppt)

			Dim updatedAppt() As CarScheduling = SchedulerExtension.GetAppointmentsToUpdate(Of CarScheduling)(SchedulerHelper.Settings, SchedulerDataHelper.GetAppointments(), SchedulerDataHelper.GetResources())
			For Each appt In updatedAppt
				SchedulerDataHelper.UpdateAppointment(appt)
			Next appt

			Dim removedAppt() As CarScheduling = SchedulerExtension.GetAppointmentsToRemove(Of CarScheduling)(SchedulerHelper.Settings, SchedulerDataHelper.GetAppointments(), SchedulerDataHelper.GetResources())
			For Each appt In removedAppt
				SchedulerDataHelper.RemoveAppointment(appt)
			Next appt
		End Sub
	End Class
End Namespace

File: CS/Controllers/HomeController.cs
using DevExpress.Web.Mvc;
using SchdulerBestPracticesMvc.Code;
using SchdulerBestPracticesMvc.Models;
using System.Web.Mvc;

namespace SchdulerBestPracticesMvc {
    public class HomeController: Controller {
        public ActionResult Index() {
            return View(SchedulerDataHelper.DataObject);
        }

        public ActionResult SchedulerPartial() {
            return PartialView("SchedulerPartial", SchedulerDataHelper.DataObject);
        }

        public ActionResult EditAppointment() {
            UpdateAppointment();
            return PartialView("SchedulerPartial", SchedulerDataHelper.DataObject);
        }

        static void UpdateAppointment() {
            CarScheduling insertedAppt = SchedulerExtension.GetAppointmentToInsert<CarScheduling>(
                SchedulerHelper.Settings,
                SchedulerDataHelper.GetAppointments(),
                SchedulerDataHelper.GetResources()
            );
            SchedulerDataHelper.InsertAppointment(insertedAppt);

            CarScheduling[] updatedAppt = SchedulerExtension.GetAppointmentsToUpdate<CarScheduling>(
                SchedulerHelper.Settings,
                SchedulerDataHelper.GetAppointments(),
                SchedulerDataHelper.GetResources()
            );
            foreach (var appt in updatedAppt) {
                SchedulerDataHelper.UpdateAppointment(appt);
            }

            CarScheduling[] removedAppt = SchedulerExtension.GetAppointmentsToRemove<CarScheduling>(
                SchedulerHelper.Settings,
                SchedulerDataHelper.GetAppointments(),
                SchedulerDataHelper.GetResources()
            );
            foreach (var appt in removedAppt) {
                SchedulerDataHelper.RemoveAppointment(appt);
            }
        }
    }
}